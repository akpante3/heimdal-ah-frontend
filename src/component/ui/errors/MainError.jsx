import React from 'react';

const MainError = ({ errors }) => (
  <div className="col-md-12 text-danger text-center">
    {errors && errors.mainError && <span className="">{errors.mainError}</span>}
  </div>
);

export default MainError;
