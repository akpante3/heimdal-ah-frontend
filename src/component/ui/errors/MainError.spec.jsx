/* eslint-disable no-undef */
import React from 'react';
import MainError from './MainError';

describe('The MainError Component Test Suite', () => {
  it('should render the MainError component', () => {
    const wrapper = shallow(<MainError />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <MainError /> Component', () => {
    const wrapper = shallow(<MainError />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.find('div')).toBeDefined();
  });

  it('should render exactly one span html tag if an error object and a name is passed into the component', () => {
    const errors = {
      mainError: 'The username is invalid',
    };
    const wrapper = shallow(<MainError errors={errors} />);
    expect(wrapper.find('span')).toHaveLength(1);
  });
});
