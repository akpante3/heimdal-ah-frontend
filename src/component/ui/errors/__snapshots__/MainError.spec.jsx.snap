// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MainError Component Test Suite should render the MainError component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainError
    errors={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className=""
      />,
      "className": "col-md-12 text-danger text-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className=""
        />,
        "className": "col-md-12 text-danger text-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
